justifi-tech/web-component-library
# Checkout customer readiness

Links
-----



Developer considerations
--------------

- On any task
  - [ ] Perform dev QA
- On new features
- On bugs


Developer QA steps
--------------------
- Use the sample app provided via Slack. Do not immediately run `npm install`.
- Run this branch locally, and from the `packages/webcomponents` directory run `npm link`
- From the sample app, run `npm link @justifi/webcomponents`
- Run `npm install` and `node index.js` to start the app
- [x] Do a payment using a saved payment method
- [x] Do a payment using a new card
- [x] Do a payment using a new bank account
- [x] `submitted` event gets emitted upon submit
- [x] `error` event upon fetch error (try disabling internet connection in dev tools)
- [x] Attempt a payment with invalid form data for card and bank account
- [x] QA feedback - Make sure that when client-side errors happen on the "new payment method" forms (leave things blank that are required) the "Pay" button does not get stuck in a loading/submitting state
- [x] QA feedback - Make sure that a new payment method can be successfully used and tokenized. For this use card # `4242424242424242`. Check your console, the `'submitted'` event should be emitted, and the `detail` should include the full success response which fits the shape of `{ data: { checkout: {...} }`
- [x] Make sure that when a new payment method is not successfully tokenized that an error event gets emitted / logged in the console. For this use card # `4000000000000002`. The error event should contain a `generic_decline` code
